### ISSUE TITLE
Unexpected behavior of auto-complete with multiple cursors

### ISSUE THREAD

**Issue by pidgeon777** [jcaillon/3P#246]
When using a single cursor, the auto-complete function is working as expected. The problem arises when multiple cursors are being used (for example in column mode):

![pniiotoxdd](https://user-images.githubusercontent.com/13739892/50074411-219ed100-01dc-11e9-9b07-5762f33c5c6b.gif)

E.g. it seems that when typing a row of characters like this:

e
e
e
e

the auto-complete function scans for words made by at least four "e" (four is equal to the number of active cursors).

Moreover, in case of a big number of multiple cursors, the auto-complete form is subject to a noticeable flickering, as is it was called in a loop for each of the cursors:

![wl6r6pzvbp](https://user-images.githubusercontent.com/13739892/50075583-a50df180-01df-11e9-91f3-8e81831f6c20.gif)

In my opinion the correct behavior of the auto-completion would be to propose suggestions as if only one cursor was active, even in multi-selection mode. By doing this, flickering would also be avoided in case of a big number of multiple selections.

This is unfortunately a very annoying behavior because auto-completion is almost unusable when recurring to multi-selection, which is used a lot. Do you think this could be classified as bug to be fixed?

Software/OS:

Notepad++ v7.5.9 32bit
3P - Progress Programmers Pal v1.8.3
Windows 10 x64

Thank you

**Comment by jcaillon** [comment]
Thank again for the great issue. This is definitely a bug and you are right on the diagnostic! Will fix asap, I will publish a build artifact in this issue so you can get a fix quickly.

**Comment by jcaillon** [comment]
Hello again, you can try the fixed version here:

https://ci.appveyor.com/api/buildjobs/t8uwum8c4qdreso0/artifacts/Releases%2F3P.zip

**Comment by pidgeon777** [comment]
Thanks for the prompt reply and fix. Well, I played a bit with your latest release, and so far everything seems to works. I only encountered a couple of crashes in Notepad++ but I would not consider your plugin to be necessarily responsible for that. Moreover, a few times the suggestion box didn't appear or placed the selected item in wrong points (different from the ones originally pointed by my multiple selections), but I will do more tests and keep you updated about any outcome. In the meanwhile, thank you again, I wish you a successful new year.
